superbench/benchmarks/micro_benchmarks/directx_gemm_flops_performance/GPUCore.cpp

```diff
@@ -0,0 +1,507 @@
+
+    if (temporaryResourceSize != 0) {
+        DML_BUFFER_BINDING bufferBinding{temporaryBuffer.Get(), 0, temporaryResourceSize};
+        DML_BINDING_DESC bindingDesc{DML_BINDING_TYPE_BUFFER, &bufferBinding};
+        m_bindingTable->BindTemporaryResource(&bindingDesc);
+    }
+
+    if (persistentResourceSize != 0) {
+        DML_BUFFER_BINDING bufferBinding{persistentBuffer.Get(), 0, persistentResourceSize};
+        DML_BINDING_DESC bindingDesc{DML_BINDING_TYPE_BUFFER, &bufferBinding};
+        m_bindingTable->BindPersistentResource(&bindingDesc);
+    }
+
+    CloseExecuteResetWait();
+
+    DML_BUFFER_BINDING inputBufferBindingA{m_inputBufferA.Get(), 0, sizeof(T) * m * k};
+    DML_BINDING_DESC inputBindingDescA{DML_BINDING_TYPE_BUFFER, &inputBufferBindingA};
+
+    DML_BUFFER_BINDING inputBufferBindingB{m_inputBufferB.Get(), 0, sizeof(T) * n * k};
+    DML_BINDING_DESC inputBindingDescB{DML_BINDING_TYPE_BUFFER, &inputBufferBindingB};
+
+    DML_BUFFER_BINDING bufferBinding = {nullptr, 0, 0};
+    DML_BINDING_DESC inputBindingDesc{DML_BINDING_TYPE_NONE, &bufferBinding};
+
+    std::array<DML_BINDING_DESC, 3> inputBindings = {inputBindingDescA, inputBindingDescB, inputBindingDesc};
+    m_bindingTable->BindInputs(3, inputBindings.data());
+
+    DML_BUFFER_BINDING outputBufferBinding{m_outputBuffer.Get(), 0, sizeof(T) * n * m};
+    DML_BINDING_DESC outputBindingDesc{DML_BINDING_TYPE_BUFFER, &outputBufferBinding};
+
+    m_bindingTable->BindOutputs(1, &outputBindingDesc);
+
+    // Execute the compiled GEMM operator and record the GPU time.
+    this->gpuTimer.start(m_commandList.Get(), 0);
+    m_dmlCommandRecorder->RecordDispatch(m_commandList.Get(), m_dmlCompiledOperator.Get(), m_bindingTable.Get());
+    this->gpuTimer.stop(m_commandList.Get(), 0);
+    this->gpuTimer.resolveQueryToCPU(m_commandList.Get(), 0);
+    CloseExecuteResetWait();
+
+    double timeInMs = this->gpuTimer.getElapsedMsByTimestampPair(0);
+
+    if (_PRINT_RESULT) {
+        // The output buffer now contains the result of the identity operator,
+        // so read it back if you want the CPU to access it.
+
+        m_commandList->ResourceBarrier(
+            1, get_rvalue_ptr(CD3DX12_RESOURCE_BARRIER::Transition(
+                   m_outputBuffer.Get(), D3D12_RESOURCE_STATE_UNORDERED_ACCESS, D3D12_RESOURCE_STATE_COPY_SOURCE)));
+
+        m_commandList->CopyResource(m_readBackBuffer.Get(), m_outputBuffer.Get());
+
+        CloseExecuteResetWait();
+        D3D12_RANGE tensorBufferRange{0, static_cast<SIZE_T>(sizeof(T) * n * m)};
+        T *outputBufferData{};
+        ThrowIfFailed(m_readBackBuffer->Map(0, &tensorBufferRange, reinterpret_cast<void **>(&outputBufferData)));
+        std::wstring outputString = L"output tensor: ";
+        for (size_t tensorElementIndex{0}; tensorElementIndex < static_cast<SIZE_T>(m * n);
+             ++tensorElementIndex, ++outputBufferData) {
+            outputString += std::to_wstring(*outputBufferData) + L' ';
```

Comment:
>to_wstring [](http://example.com/codeflow?start=33&length=10)

two questions here: 1) why we use wstring instead of string here? 2) can outputBufferData be transferred to a string? I suppose it should be some binary data.